src/org/rascalmpl/util/maven/SimpleResolver.java

```diff
@@ -149,12 +164,12 @@ private void downloadJar(Path local, String groupId, String artifactId, String v
     private void downloadArtifact(Path local, String groupId, String artifactId, String version, boolean force) throws UnresolvableModelException {
         // TODO: deal with repository filters etc in settings.xml
         var url = getUrl(local, groupId, artifactId, version);
-        for (var r : availableRepostories) {
-            if (r.repo.getLayout().equals("legacy")) {
+        for (SimpleRepositoryDownloader repoDownloader : availableRepostories) {
```

Comment:
nit: `SimpleRepositoryDownloader repoDownloader`  is twice the same text. so `var` wouldn't loose information, but would make the code smaller (ie, faster to read?) (I mean, I think that `repoDownloader` is indeed a better name than `r`)